principles/coding-principles.md

```diff
@@ -63,24 +63,32 @@ The larger the chunk of work you’re estimating, the more inaccurate you will b
-  
-### 2. Use static code analysis
-To ensure that code meets the quality criteria of the Apprenticeship Service static code analysis via SonarCloud.io should be utilised and integrated into the build pipelines. The output of this analysis should be reviewed as part of the Code Review process and any issues rectified
 
-## Testing-Specific Principles
+### 2. SOLID
+Follow [SOLID principles](https://en.wikipedia.org/wiki/SOLID). This will make code more understandable and easier to maintain
+* Single Responsibility - 
+* Open/Closed principle
+* Liskov Substitution
+* Interface Segregation
+* Dependency Inversion
 
-### 1. Write tests
+### 3. Use static code analysis
+To ensure that code meets the quality criteria of the Apprenticeship Service static code analysis via SonarCloud.io should be utilised and integrated into the build pipelines. The output of this analysis should be reviewed as part of the Code Review process and any issues highlighted should be rectified
 
-Always write tests for any code that you write. Besides giving developers confidence that the code works, the tests will give context that otherwise would be missing. This is especially valuable a few years down the road when all of the original developers have left the project.
+### 4. Code Reviews
+Ensure reviews are carried out on all code regardless of where the change originates. Also ensure the process is carried out with the appropriate riger and attention. Some things to keep in mind during the code review process:
 
-### 2. Use test-driven development or behaviour-driven development
-Use test-driven development or [Behaviour Driven Development (BDD)](http://dannorth.net/introducing-bdd/) if possible. It ensures that before you write your code that you have success criteria, and that your code is testable.
+* Give reviews your full attention
+* Limit the amount of time you spend reviewing in any one sitting
+* If necessary check the code out and test it locally
+* Give constructive feedback
+* Consider including non-developers in the review process, they may pick up different issues
+* Do not merge code when there are unresolved comments or requests for change
 
-### 3. Follow the testing pyramid
-The makes the point to write many low level unit tests, fewer integration tests, and even fewer acceptance tests
-![testing pyramid](./images/test-pyramid.png)
+### 5. Test Driven Development
+Use test-driven development or [Behaviour Driven Development (BDD)](http://dannorth.net/introducing-bdd/) principles. It ensures that before you write your code that you have success criteria, and that your code is testable. It gives developers confidence that the code they have written works and the tests will give context that otherwise would be missing, this is especially valuable when the original developers have left the project.
 
```

Comment:
You've deleted the 'Follow the testing pyramid' section. I think its worth keeping in. Was it an intentional deletion?

Reply:
It is now covered in the testing principles where I believe it more logically sits